nav2_behavior_tree/include/nav2_behavior_tree/navigate_to_pose_action.hpp

```diff
@@ -41,8 +43,8 @@ class NavigateToPoseAction : public BtActionNode<nav2_msgs::action::NavigateToPo
     geometry_msgs::msg::Point position;
     geometry_msgs::msg::Quaternion orientation;
 
-    bool have_position = getParam<geometry_msgs::msg::Point>("position", position);
-    bool have_orientation = getParam<geometry_msgs::msg::Quaternion>("orientation", orientation);
+    bool have_position = getInput("position", position);
```

Comment:
Nit pick, do or don't, but since we dont use the `have_*` results, you could reformulate to:
```
if (!getInput(...) || !getInput(...))
```

Downside is that then it won't  get to the second condition if the first fails - but in that case wouldn't it be fine to return both empty? Not sure if the intent after the ERROR log should really be to return and not set `goal_` at all

Reply:
Let me defer this change. So near as I can tell, we should be able to completely delete the checks for these parameters since we provide default values. However, that doesn't seem to work in practice. I'm not sure if I'm doing something wrong or if there is a bug in the library.